src/cs/Ssh/SecureStream.cs

```diff
@@ -29,6 +29,7 @@ public class SecureStream : Stream
 	private readonly SshClientCredentials? clientCredentials;
 	private readonly SshServerCredentials? serverCredentials;
 	private SshStream? stream;
+	private readonly TaskCompletionSource<bool> connectCompletion = new ();
```

Comment:
Is the `bool` result of `connectCompletion` used? Can it be just `TaskCompletionSource` ?

Reply:
The non-generic `TaskCompletionSource` is only available starting with .NET 5. But this project still maintains compatibility with .NET Framework.